pkg/controller/apiextensions/composite/reconciler.go

```diff
@@ -265,43 +260,65 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	// array with the same length. Then copy the already provisioned ones into
 	// that array to not create new ones because composed reconciler assumes that
 	// if the reference is empty, it needs to create the resource.
+
+	// TODO(negz): This approach means that the resources of a Composition are
+	// effectively append only. We may want to reconsider this per
+	// https://github.com/crossplane/crossplane/issues/1909
 	refs := make([]corev1.ObjectReference, len(comp.Spec.Resources))
```

Comment:
Not related to this PR, but we should get rid of `ObjectReference` here due to https://github.com/crossplane/crossplane-runtime/issues/49

Reply:
Yeah, I'm definitely keen to do that. I suspect it might be best done in its own PR though - I considered doing it in this one for a split second but decided it would add too much unrelated churn.